### ISSUE TITLE
[html] PMD html XPATH false positive

### ISSUE THREAD

**Issue by codefriar** [pmd/pmd#4745]
**Affects PMD Version:** 
 7.0.0-rc4

**Description:**

An XPath selector has been developed that behaves differently in PMD than other Xpath evaluation tools. 

Given this HTML Snippet

```html
<table>
  <tr>
    <th>Company</th>
    <th>Contact</th>
    <th>Country</th>
    <th></th>
  </tr>
  <tr>
    <td>Alfreds Futterkiste</td>
    <td>Maria Anders</td>
    <td>Germany</td>
  </tr>
  <tr>
    <td>Centro comercial Moctezuma</td>
    <td>Francisco Chang</td>
    <td>Mexico</td>
  </tr>
</table>
```

And the xpath selector 

```xpath
//th[not(normalize-space())]
```

PMD, and the designer, wrongly select all four `<th>` elements, even though only the last one is missing a text name.

Other xpath testing tools such as xpather.com and Chrome's console both work as-expected, identifying only the 4th <th> element.

**Steps to reproduce:**

Open pmd designer.
open an html file with the above html template code in it.
enter this selector

```xpath
//th[not(normalize-space())]
```

see it identify all 4 `<th>` elements.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

This happens in PMD Designer as well as the CLI `pmd check`

**Comment by codefriar** [comment]
Interestingly enough, the following selector also false-positives. It will identify all `<th>` elements. 

```xpath
//th[string-length(normalize-space()) < 1]
```

This selector does work correctly in chrome and xpather.com

**Comment by codefriar** [comment]
This selector, however, *does work* 

```xpath
//th[not(node()) and not(normalize-space())]
```

**Comment by adangel** [comment]
Please see https://docs.pmd-code.org/latest/pmd_languages_html.html

> XPath 2.0 rules are supported, but the DOM is not always a typical XML/XPath DOM. In the Designer, text nodes appear as nodes with name “#text”, but they can be selected as usual using text().

So, your XPath should look like this:

    //th[not(normalize-space(text()))]


However, maybe this behavior can indeed be improved: https://github.com/pmd/pmd/blob/1c8c5cf4343b406168d27740f34c6f58cc6369d8/pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/xpath/internal/AstElementNode.java#L224
